test/specs/configs/full.yaml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2018 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+# defines the default content repository
+# content: local
+content: https://github.com/adobe/project-helix.io/blob/master
+
+# defines the default code repository
+code: https://github.com/adobe/project-helix.io/blob/master
+
+# defines the default static root relative to the code repository
+staticRoot: /webroot
+
+# defines the directory index file
+directoryIndex: index.html
+
+# define additional configs (strains)
+strains:
+  default:
+    code: /acapt/default/https---github-com-adobe-project-helix-io-git--master--
```

Comment:
@kptdobe I copied this from project-helix.io. what is this `code` for? I think it should point to a git repository and not a resolved action. the action resolution should be done during `hlx publish` no ?

Reply:
can we call it `actionBase` or `actionRoot` or `actionPrefix` instead? 

otoh, I don't really like that `hlx deploy` modifies the strains config. I think it is only needed, so that `hlx publish` knows that to put into the VCL. so maybe we do:

1. `hlx deploy` reads the `helix-config.yaml` -> `HelixConfig`
2. `hlx deploy` updates the strains in `HelixConfig.strains()` with the action roots
3. `hlx deploy` writes a fully resolved `.hlx/build/strains.json`<sup>1<sup> using `HelixCondig.writeStrains()`
4. `hlx publish` uses the `.hlx/build/strains.json` to update the VCL.

<sup>1<sup> We choose json here to discourage editing :-)

Reply:
I'm ok with `action`.

Reply:
Would `.hlx/build/strains.json` be under source control or in `.gitignore`?

Reply:
Do we need additional options in `hlx deploy` that attach a deployed branch to a particular strain? Something like `hlx deploy --attach preview`?

Reply:
> Would .hlx/build/strains.json be under source control or in .gitignore?

no.